scripts/coverage_report_jenkins.py

```diff
@@ -51,6 +118,224 @@ def get_build_numbers(client, job_name):
     return [build['number'] for build in client.get_job_info(job_name)['builds']]
 
 
+def merge_coverage_data(ssh, coverage_dir):
+    """Merge coverage data
+
+    Take all the by appliance by process .resultset.json files from
+    the coverage archive and merge them into one .resultset.json file.
+    Expects the coverage archive to have been extracted to the
+    coverage_dir on the appliance to which the ssh client is connected.
+
+    Args:
+        ssh:  ssh client
+        coverage_dir:  Directory where the coverage archive was extracted.
+
+    Returns:
+        Nothing
+    """
+    print('Merging coverage data')
+
+    # Run the coverage merger script out of the rails root pointing
+    # to where the coverage data is installed.   This will generate
+    # a directory under the coverage directory called merged, and
+    # will have the merged .resultset.json file in it, along with some
+    # HTML that was generated by the merger script.
+    cmd = ssh.run_rails_command(
+        'coverage_merger.rb --coverageRoot={}'.format(coverage_dir),
+        timeout=60 * 60)
+    if not cmd:
+        print('Failure running the merger - {}'.format(str(cmd)))
+        return 6
+    else:
+        print('Coverage report generation was successful')
+        print(str(cmd))
+        percentage = re.search(r'LOC\s+\((\d+.\d+%)\)\s+covered\.', str(cmd))
+        if percentage:
+            print('COVERAGE={};'.format(percentage.groups()[0]))
+        else:
+            print('COVERAGE=unknown;')
+
+    # The sonar-scanner will actually need the .resultset.json file it
+    # uses to be in /coverage/.resultset.json (i.e. the root of the coverarage
+    # directory), so lets create a symlink:
+    ssh.run_command('ln -s merged/.resultset.json {}/.resultset.json'.format(coverage_dir))
+
+
+def pull_merged_coverage_data(ssh, coverage_dir):
+    """Pulls merged coverage data to log directory.
+
+    Args:
+        ssh:  ssh client
+        coverage_dir:  Directory where the coverage archive was extracted.
+
+    Returns:
+        Nothing
+    """
+    print('Packing the generated HTML')
+    cmd = ssh.run_command('cd {}; tar cfz /tmp/merged.tgz merged'.format(coverage_dir))
+    if not cmd:
+        print('Could not compress! - {}'.format(str(cmd)))
+        return 7
+    print('Grabbing the generated HTML')
+    ssh.get_file('/tmp/merged.tgz', log_path.strpath)
+    print('Decompressing the generated HTML')
+    rc = subprocess.call(
```

Comment:
Checkout `subprocess.check_call()` here, or `subprocess.check_output()` if you want the command output captured.  They will do the RC=0 check for you, and raise CalledProcessError if it fails.  That way this can be a bit more pythonic, looking like:

```python
try:
    subprocess.check_call(['tar', 'stuff'])
    logger.info('Done!')  # or just print :)
except CalledProcessError:
    logger.error('it broke')
    return

Reply:
And in my case I don't even need the try catch, because the exception raised in that case is what i want if it fails.